sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1040,6 +1079,17 @@ static int get_slave_info(const struct snd_soc_acpi_link_adr *adr_link,
 	return 0;
 }
 
+static void set_dailink_map(struct snd_soc_dai_link_codec_ch_map *sdw_codec_ch_maps,
+			    int codec_num, int cpu_num)
+{
+	int step;
+	int i;
+
+	step = codec_num / cpu_num;
```

Comment:
This assumes cpu_num and codec_num are integer multiples, so we should enforce that at the framework level.

Reply:
> This assumes cpu_num and codec_num are integer multiples, so we should enforce that at the framework level.

@plbossart My thought is that we keep the flexibility at the framework level, and machine driver can set the mapping as they need. Here I set connected_cpu_id like that because it fits our current configurations. However, people still can set connected_cpu_id to fit any HW configurations. For example, say there are 3 codecs and 2 cpus, and codec A is connected to cpu 1 and codec B and C are connected to cpu 2.